### PR TITLE
Add standard constant strings for CT parsing

### LINKED ISSUE #5: Add constants to maeparser headers
When integrating maeparser with RDKit, @mbeachy raised the point that we should probably move string constants like 'f_m_ct' into headers in maeparser, where we can name them in obvious ways that make them look less-magicky in lilbrary-user code.  The obvious things we could constant:

- f_m_ct
- s_m_title
- m_atom
- r_m_*_coord
- m_bond
- i_m_from / i_m_to / i_m_order
- i_m_formal_charge (maybe some other common properties?)

### PR DESCRIPTION
**Pull request by lorton**
Switched our tests and usage demo to use these constants.  Updated
README.md to inform of their existence.

fixes #5

### CODE REVIEW

**Comment by lorton** [comment]
@speleo3 - mind reviewing this one for me @speleo3  ?

File: MaeConstants.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+const std::string MAE_FORMAT_VERSION = "s_m_m2io_version";
+const std::string CT_BLOCK = "f_m_ct";
+const std::string CT_TITLE = "s_m_title";
+const std::string ATOM_CT_BLOCK = "m_atom";
+const std::string ATOM_ATOMIC_NUM = "i_m_atomic_number";
+const std::string ATOM_X_COORD = "r_m_x_coord";
+const std::string ATOM_Y_COORD = "r_m_y_coord";
+const std::string ATOM_Z_COORD = "r_m_z_coord";
+const std::string ATOM_CHARGE = "i_m_formal_charge";
+const std::string BOND_CT_BLOCK = "m_bond";
```
Comment (torcolvin) [inline MaeConstants.hpp]: just call BOND_BLOCK

File: MaeConstants.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+const std::string BOND_FROM_ATOM = "i_m_from";
+const std::string BOND_TO_ATOM = "i_m_to";
```
Comment (torcolvin) [inline MaeConstants.hpp]: I am tempted to use index1 or index2 (or index a and b) and drop the to and from concept. We don't support bond directionality and we may or may not have both bond directories in the file.

**Comment by lorton** [comment]
Any more thoughts on this @torcolvin ?  This can't be _that_ strange of an ask, should I just investigate how other libraries that use CMake do this?

File: test/ReaderTest.cpp [hunk 3]
```diff
@@ -31,7 +32,7 @@ BOOST_AUTO_TEST_CASE(Reader0)
 
     auto b = r.next("");
     BOOST_REQUIRE(b);
-    BOOST_REQUIRE_EQUAL(b->getStringProperty("s_m_m2io_version"), "1.1.0");
+    BOOST_REQUIRE_EQUAL(b->getStringProperty(MAE_FORMAT_VERSION), "1.1.0");
```
[the remark this answers is not in the document]
Comment (lorton) [reply test/ReaderTest.cpp]: This format is only (and will only ever be) used for Maestro-based files, so adding the notion of M2IO is just unnecessarily confusing for people.

**Comment by lorton** [comment]
@speleo3 - I'm curious what you mean by "introduces a new alias that the user must learn about" - no matter what we write there will be an alias the user has to learn.  Are you suggesting it's better to use `i_m_atomic_number` because it's something the user doesn't need to learn about if they're working close to the 'data spec'?  What exactly are you suggesting here? :)

**Comment by speleo3** [comment]
I'm saying that you actually have to look at the values of those constants if you want to use them. Sure, most of them are obvious (can't really misinterpret `ATOM_X_COORD`), but e.g. `ATOM_CHARGE` could be `i_m_formal_charge`, `r_m_charge1`, or possibly something else. The user (programmer) likely already knows what's inside an .mae file, either by opening it in a text editor, or in Maestro and displaying atom properties etc. Looking up aliases for the data names he finds there isn't hard, but doesn't add value either in my opinion.

**Comment by lorton** [comment]
Hi Thomas - here's where we disagree significantly.  I can't imagine a world where having magic strings 'f_m_atom_x' is clearer than a constant `ATOM_X_COORD`.  You raise a valid point about ATOM_CHARGE, but the point really is just that it should be changed to ATOM_FORMAL_CHARGE to get rid of the ambiguity.

I also disagree with the assertion that the user/programmer likely knows the mae format already.  That's going to be true for the tiny fraction of people who have already programmed against Maestro (it's not even in Babel!), but it's certainly not going to be true for people who are new to the format.

Most users are going to want to read the library documentation, a couple examples, and be able to access the format without knowing any specifics.  They're going to want constants, not magic strings.  I definitely expect that behavior whenever I'm working with a format and its libraries.

**Comment by speleo3** [comment]
Thanks for changing `ATOM_FORMAL_CHARGE`! Sorry for having been so critical, I can't disagree with all the advantages you point out. From my side, this is good to go.

Can you add `ATOM_PARTIAL_CHARGE = "r_m_charge1"` ?

**Comment by lorton** [comment]
Hi @speleo3 - no need to apologize, just discussing philosophy :)  I'll add partial charge.

**Comment by lorton** [comment]
Closing this to switch to a rebased pull request

**Pull request closed without merging**